### ISSUE TITLE
Sorry to say but your code is not at all clear.

### ISSUE THREAD

**Issue by rishikeshraj5** [lessonxmk/head_fusion#1]
I read the readme file, in three lines t=you have described it. You should start by explaining the original dataset files, as anybody could have saved right?
Where did you use the renamed files in train.py?
where did you save the renamed files?

Regards

**Comment by lessonxmk** [comment]
line 58 in in train.py:
WAV_PATH = "E:/Test/IEMOCAP/"

Please move  the renamed files the this path

**Comment by rishikeshraj5** [comment]
Ok sir thank you so much for this reply, we are organizing an AI workshop
here in our college , certainly going to cite your  work also, But some
how I am not able to implement it.
So I hope if this is your original work and already published you can help
our team.

Also the renamed files is json file right?
Sir this PAIR dict has files name and label right?
and this renamed file folder is moved here in train.py file right?

Uptill now,
I have changed the original files name, the PAIR dict is saved in json file
format.

apart from this whatelse is required to change please guide.

Regards

**Comment by lessonxmk** [comment]
I have been busy lately and may not always reply in time.

The pair.json is not important.
It just helps the work of renaming.
The handleIEMOCAP.py is used to rename all wav files like this formant: "Ses01F-impro01-XX-01-F000.wav" （We used the RAVDESS data set at the beginning, this format is compatible with the file name of the RAVDESS data set.）

This format is used when generating train and test data set in line 111 in train.py.

**Comment by rishikeshraj5** [comment]
Sir here in lineos.rename(_, dst='name')
can I change the dst location actually I don't want to change the original files name. 
This code is changing the files names there in my original data, Plz guide

**Comment by lessonxmk** [comment]
Why not back up the original files？

**Comment by rishikeshraj5** [comment]
Sir thats whole dataset, but if there is no other way around I hv to do it
only

**Comment by lessonxmk** [comment]
Although I think the whole data set is not very large, if this is a bit difficult for you, you can try to backup only the wav files, they are only about 1GB.

**Comment by rishikeshraj5** [comment]
Okay sir, whole data is about 24.5 gb .

**Comment by rishikeshraj5** [comment]
So sir, I ran this handleIEMOCAP.py file ,
My data files names are changed now. So in line 58 or train.py I have
changed the location as PATH_WAV in  ''handleIEMOCAP.py file''.......right?

Then finally training is required. But I where is these features are
calculated by features.py, I think using get_mfcc, But how this file
named 'features_mfcc_impro.pkl' is created at line 56.

**Comment by lessonxmk** [comment]
Please set _featuresExist_ on line 54 to False.
See line 205, when _featuresExist_ is false, the feature will be extracted and saved as .pkl file

**Comment by rishikeshraj5** [comment]
Yes sir did that already, Actually I did read the code carefully, But I
don't know why its saying

IndexError: list index out of range
at line

  label = str(os.path.basename(wav_file).split('-')[2])

does it mean wav file name doesn't have index [2] . file file path is

'E:/dataset/data/IEMOCAP_full_release\\Session1\\sentences\\wav\\Ses01F_impro01\\Ses01F-impro01-01-01-F001.wav'

then basefile name is  Ses01F-impro01-01-01-F001.wav right sir?
Then why such error.
Is it related to next lines in code coz when I ran it with same error line
by line its showing label ='01'

STUCK UP
I have tomorrow in my hand only to train this model sir.

**Comment by rishikeshraj5** [comment]
Sir its working when I checked line by line for single wave file then why
this error any idea sir?

**Comment by rishikeshraj5** [comment]
Sir plz help I am getting this trackback of error


 0%|          | 6/8030 [00:00<02:35, 51.60it/s]constructing meta dictionary
for E:/dataset/data/IEMOCAP_full_release/*/sentences/wav/*/S*.wav...
 11%|█         | 848/8030 [00:08<01:07, 105.75it/s]
Traceback (most recent call last):

  File
"E:\Python_On_All_Dataset\IEMO\long_codes\multiscale_att_impro_only\head_fusion-master\train.py",
line 215, in <module>
    train_X, train_y, val_dict = process_data(WAV_PATH, t=T_stride,
train_overlap=T_overlop)

  File
"E:\Python_On_All_Dataset\IEMO\long_codes\multiscale_att_impro_only\head_fusion-master\train.py",
line 113, in process_data
    label = str(os.path.basename(wav_file).split('-')[2])

IndexError: list index out of range



Sir this is working then y it stopped in between?
plz guide

**Comment by lessonxmk** [comment]
Please check wav file names. Whether there are some files renamed incorrectly and the file name can not be split by '-' ?

**Comment by rishikeshraj5** [comment]
No sir all files names are absolutely correct with '-' and every third
segment label like 01,02,,09.
Sir plz brief me about your data arrangement I think there is some problem.
In my original dataset the path for wav files is :
"E:/dataset/data/IEMOCAP_full_release/*/sentences/wav/*/S*.wav"
using handleIEMOCAP.py i have changed all files names in this folder only.
Then I replaced the path in line 58 as
WAV_PATH = "E:/dataset/data/IEMOCAP_full_release/*/sentences/wav/*/S*.wav"

is there any mistake sir?

In your readme file here in second step you are saying move renamed files
to dict, what does it mean sir?



1.run handleIEMOCAP to rename files in IEMOCAP corpus.

*2.move the renamed files to a dictionary (e.g. '/data/*.wav').*

3.run train.py to train a model and evaluate.

**Comment by lessonxmk** [comment]
The second step means moving all the renamed files to an independent folder. I am not sure if it is correct to use regular expressions in  WAV_PATH.

And WAV_PATH should be folder name instead of file name.
For example, there are some files (*.wav) in 'E:/dataset/' , WAV_PATH should be 'E:/dataset/' instead of 'E:/dataset/*.wav'

Finally, you can use debug mode in your IDE to find out which file it is when this error occurred.

**Comment by rishikeshraj5** [comment]
I think the error is in the folder path itself, as you are saying all renamed files in a single folder (all 10000 renamed files in a single folder). so there should not be any subfolders??
right? But sir that's really tough as, there are almost 30 files inside each session file and then inside each such files 50 to 100 wav files, so unfolding each of these almost 10000 files that's really time-consuming. 


and yes that's folder name only as I have changed the third line in
process_data definition to

wav_files = glob.glob(path)
instead of file extension.

Plz reply so that I can go with the unfolding of these 10k files in a single folder sir.

**Comment by rishikeshraj5** [comment]
sir could you plz tell me if that can work in this situation, what should
be the modification in this code then.
plz

**Comment by rishikeshraj5** [comment]
Not able to make it sir!
well could you plz tell me what is the acc you are getting from this model?
are you actually getting 76% from this model? or there are some other hyperparameters adjustments?

**Comment by lessonxmk** [comment]
Since the data set is randomly divided, the results also have some randomness. The accuracy may be a bit lower or higher, just like this figure. 76% is the average accuracy of several experiments.
![QQ截图20210526200601](https://user-images.githubusercontent.com/44159619/119657203-32000f00-be5e-11eb-87ff-7e61a124adcc.png)

**Comment by rishikeshraj5** [comment]
Thank you for the clarification. So here acc is the weighted acc at 50th
epoch not average of all 50 epochs right?
and you are considering acc after 50th epoch, I think it would be better to
see this in an loss accuracy graph for train and val data.
and sir this is improvised data only for four emotional classes right ie
2943 utterances as in paper you merged happy and exc or some other
combination?

**Comment by rishikeshraj5** [comment]
Sir in this screen shot , The maxUA and maxWA should be the final result I
think, so in this case 78% NO? Plz correct me If m wrong?
Also for three versions of datasets impro , script and all
do we need to change anything else in code, because while enumerating the
train and val files there are some conditions in dark fonts :

 for i, wav_file in enumerate((valid_files)):
        label = str(os.path.basename(wav_file).split('-')[2])
        if (label not in LABEL_DICT1):
            continue

* if (impro_or_script != 'all' and (impro_or_script not in wav_file)):
      continue*
        label = LABEL_DICT1[label]
        wav_data, _ = librosa.load(wav_file, sr=RATE)

**Comment by XYQ159371** [comment]
Hello, I am very sorry to disturb you. Could you please explain the source of the noise set? I would like to use it. If it is convenient for you, plz send my email. (12shq12@163.com) I would appreciate it.

**Comment by Coding511** [comment]
@XYQ159371 you can download these noise files from here dierctly.
